### PR TITLE
This is implementation of controlling auxilary PWM outputs with a mission item (mavlink message id = 183).

### PR DESCRIPTION
**Pull request by fpvaspassion**
Hello, 
Some time ago I faced a problem with controlling servos attached to PX4 during mission execution.
After some discussion here
 https://groups.google.com/forum/#!topic/px4users/_IrwFDOd2p8 
I implemented this functionality.

Regards,
Michail.

### CODE REVIEW

**Comment by fpvaspassion** [comment]
I've tested it in flight, but unfortunately I don't have any logs right now. However I don't think they would be of any use in this case.

**Comment by 89craven89** [comment]
can i merge this branch so i can give it a test?

**Comment by fpvaspassion** [comment]
Hello, yes you can do it.

**Comment by 89craven89** [comment]
betwenn when you tested did you use the custom widget  with the messase mav_cmd_do_set_servo?
because in qgroundcontrol the second parameter cant be used with more than 2 digits thus you cant set a normal pwm value to the servo

**Comment by fpvaspassion** [comment]
When testing i used it only in mission, here is the test mission to test servo on AUX6 output when flying betwen two points, first parameter is AUX chan id from 1 to 6, and second parameter is the PWM value.
QGC WPL 120

0	1	3	16	4	5	0	0	58.7106822000000008	35.828629699999999	4	1

1	0	3	183	6	900	0	0	0	0	4	1

2	0	3	16	4	5	0	0	58.7106336850431632	35.8289498448371887	4	1

3	0	3	183	6	2000	0	0	0	0	4	1

4	0	3	16	4	5	0	0	58.7106822000000008	35.828629699999999	4	1

5	0	3	183	6	900	0	0	0	0	4	1

6	0	3	16	4	5	0	0	58.7106336850431632	35.8289498448371887	4	1

7	0	3	183	6	2000	0	0	0	0	4	1

8	0	3	16	4	5	0	0	58.7106822000000008	35.828629699999999	4	1

9	0	3	183	6	900	0	0	02	0	4	1

10	0	3	16	4	5	0	0	58.7106336850431632	35.8289498448371887	4	1

11	0	3	183	6	2000	0	0	0	0	4	1

12	0	3	16	4	5	0	0	58.7106822000000008	35.828629699999999	4	1

13	0	3	183	6	900	0	0	0	0	4	1

14	0	3	16	4	5	0	0	58.7106336850431632	35.8289498448371887	4	1

15	0	3	183	6	2000	0	0	0	0	4	1

**Comment by 89craven89** [comment]
Can you provide us with the qml file for this custom command widget?

**Comment by 89craven89** [comment]
could you provide us with the qml file?

**Comment by 89craven89** [comment]
i still get the unsupported command message . I m trying to test it in manual mode , could this be the issue?

This is how i modified the qml file : 

QGCButton {
            text: "Servo Test"
            onClicked: controller.sendCommand(183, 0, 3, 5, 900, 0, 0, 0, 0, 0)
        }

**Comment by LorenzMeier** [comment]
I'm unsure about the testing state here - is this working or WIP?

File: src/modules/navigator/mission_block.cpp [hunk 1]
```diff
@@ -71,6 +74,13 @@ MissionBlock::~MissionBlock()
 bool
 MissionBlock::is_mission_item_reached()
 {
+	if (_mission_item.nav_cmd == NAV_CMD_DO_SET_SERVO) {
+
+		up_pwm_servo_arm(true);
```
Comment (LorenzMeier) [inline src/modules/navigator/mission_block.cpp]: This is crossing the abstraction layers and not something we should be generally doing. The right approach would be to publish to the actuator_control group #2 which is for payload, and to add a mixer in your airframe config which maps this to the output. The reason is that we want our architecture to scale to PWM but also UAVCAN.
https://pixhawk.org/dev/mixing#control_groups

**Comment by LorenzMeier** [comment]
You need to whitelist the mission item here now to indicate its supported:
https://github.com/PX4/Firmware/blob/master/src/modules/navigator/mission_feasibility_checker.cpp#L212

**Pull request closed without merging**